src/databricks/sqlalchemy/_types.py

```diff
@@ -0,0 +1,213 @@
+import sqlalchemy
```

Comment:
Note for reviewers: this PR renamed `types.py` → `_types.py`, which makes this file appear brand new. But most of the contents of the file are identical to `main`. Reviewing commit-by-commit you'll see the file moved in a dedicated commit. Subsequent changes to the file are linked to un-skipping the tests that those changes enabled.